src/LIBSVM.jl

```diff
@@ -420,19 +393,15 @@ function svmpredict(model::SVM{T}, X::AbstractMatrix{U}; nt::Integer = 0) where
         decvalues = zeros(Float64, dcols, ninstances)
     end
 
-    ccall((:svm_set_print_string_function, libsvm), Cvoid,
-          (Ptr{Cvoid},), @cfunction(svmnoprint, Cvoid, (Ptr{UInt8},)))
+    libsvm_set_verbose(false)
 
     cmod, data = svmmodel(model)
-    ma = [cmod]
 
     for i = 1:ninstances
```

Comment:
refine this code block 
```diff
diff --git a/src/LIBSVM.jl b/src/LIBSVM.jl
index 7f9f145..1d62294 100644
--- a/src/LIBSVM.jl
+++ b/src/LIBSVM.jl
@@ -397,19 +397,14 @@ function svmpredict(model::SVM{T}, X::AbstractMatrix{U}; nt::Integer = 0) where
 
     cmod, data = svmmodel(model)
 
-    for i = 1:ninstances
-        if model.probability
-            output = libsvm_predict_probability(cmod, nodeptrs[i], decvalues[:, i])
-        else
-            output = libsvm_predict_values(cmod, nodeptrs[i], decvalues[:, i])
-        end
-        if model.SVMtype == EpsilonSVR || model.SVMtype == NuSVR
-            pred[i] = output
-        elseif model.SVMtype == OneClassSVM
-            pred[i] = output > 0
-        else
-            pred[i] = model.labels[round(Int,output)]
-        end
+    predf = ifelse(model.probability, libsvm_predict_probability, libsvm_predict_values)
+    decode = model.SVMtype ∈ (EpsilonSVR, NuSVR) ? identity :
+             model.SVMtype == OneClassSVM        ? >(0)     :
+             (x -> model.labels[round(Int, x)])
+    f = decode ∘ predf
+
+    for i ∈ Base.OneTo(ninstances)
+        pred[i] = f(cmod, nodeptrs[i], decvalues[:, i])
     end
 
     (pred, decvalues)

```

Reply:
Nice. Would you mind turning this into a separate commit? I have a few suggestions that we could discuss there.

Reply:
sure,  it's here: https://github.com/JuliaML/LIBSVM.jl/pull/77